source/adapters/hip/memory.cpp

```diff
@@ -268,30 +225,16 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemGetInfo(ur_mem_handle_t hMemory,
 /// \param[out] phNativeMem Set to the native handle of the UR mem object.
 ///
 /// \return UR_RESULT_SUCCESS
-UR_APIEXPORT ur_result_t UR_APICALL
-urMemGetNativeHandle(ur_mem_handle_t hMem, ur_native_handle_t *phNativeMem) {
-#if defined(__HIP_PLATFORM_NVIDIA__)
-  if (sizeof(BufferMem::native_type) > sizeof(ur_native_handle_t)) {
-    // Check that all the upper bits that cannot be represented by
-    // ur_native_handle_t are empty.
-    // NOTE: The following shift might trigger a warning, but the check in the
-    // if above makes sure that this does not underflow.
-    BufferMem::native_type UpperBits = std::get<BufferMem>(hMem->Mem).get() >>
-                                       (sizeof(ur_native_handle_t) * CHAR_BIT);
-    if (UpperBits) {
-      // Return an error if any of the remaining bits is non-zero.
-      return UR_RESULT_ERROR_INVALID_MEM_OBJECT;
-    }
-  }
-  *phNativeMem = reinterpret_cast<ur_native_handle_t>(
-      std::get<BufferMem>(hMem->Mem).get());
-#elif defined(__HIP_PLATFORM_AMD__)
-  *phNativeMem = reinterpret_cast<ur_native_handle_t>(
-      std::get<BufferMem>(hMem->Mem).get());
-#else
-#error("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
-#endif
-  return UR_RESULT_SUCCESS;
+UR_APIEXPORT ur_result_t UR_APICALL urMemGetNativeHandle(ur_mem_handle_t,
+                                                         ur_native_handle_t *) {
+  // FIXME: there is no good way of doing this with a multi device context.
+  // If we return a single pointer, how would we know which device's allocation
+  // it should be?
+  // If we return a vector of pointers, this is OK for read only access but if
+  // we write to a buffer, how would we know which one had been written to?
+  // Should unused allocations be updated afterwards? We have no way of knowing
+  // any of these things in the current API design.
+  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
I want to avoid using `UR_RESULT_ERROR_ADAPTER_SPECIFIC` which might be used to encapsulate more information about the error, since the mechanism that it uses is unintuitive, and it is likely getting refactored soon. See here https://github.com/oneapi-src/unified-runtime/issues/762 . I think it's OK to leave this as unsupported since every adapter is free to choose which UR entry points it implements

Reply:
👍